### ISSUE TITLE
Magiclysm weapons missing weapon flags

### ISSUE THREAD

**Issue by Ithilrandir** [CleverRaven/Cataclysm-DDA#55717]
### Describe the bug

Weapons from magiclysm do not have the corresponding flag for its type and thus cannot be used with the martial arts for that weapon.

### Steps to reproduce

1.Learn a martial art which uses a weapon (i.e setting sun for quarterstaves)
2. Select the martial art style
3. Equip magus staff (any of the staves in magiclysm will work as well)

### Expected behavior

The game will say that the weapon is not a valid weapon type for the martial art

### Versions and configuration

Cataclysm-DDA experimental build 2022-02-26-0227

### Additional context

Some weapons seems to have their flags properly like the Orichalcum longsword.

**Comment by trip-flip** [comment]
A part of the problem lies not just within the weapon, but the actual art itself.

Here are the weapons that can work with the "Setting Sun" martial art, with no exception:
```json
    "weapons": [
      "i_staff",
      "kris",
      "kris_fake",
      "shock_staff",
      "sword_crude",
      "sword_nail",
      "sword_wood",
      "sword_xiphos",
      "q_staff",
      "wakizashi",
      "wakizashi_inferior",
      "wakizashi_fake"
    ],
```
If you take another mod weapon from the same mod, the "quarterstaff +1", and try to use it, it fails, since it's id (`q_staff_plus_one`) isn't in this list. 

We can just `copy-from` the quarterstaff onto the "Magus Staff" (since it *is* a quarterstaff), but the martial art itself needs to be fixed like so:
```json
"weapon_category": [ "QUARTERSTAVES" ]
```
This doesn't include the other weapon categories that also need to be added. Other martial arts may be need to be updated too, and other weapons may need tweaking so they work properly.

**Comment by catdach** [comment]
Basically what trip-flip said. 

Right now weapon categories for mods are still a work in progress. pretty much all the vanilla martial arts _should_ work with _most_(some still need to be categorized) mod weapons that are based on vanilla items (via weapon category inheritance). However martial arts from mods (setting sun is from MMA), still need weapon categories.

TLDR: soon™️
